### PR TITLE
Added fields to filterdialog for results page.

### PR DESCRIPTION
**Pull request by saberlynx**
Added filter keywords owner, host, cvss base score, vulnerability and location to filter dialog on results page. Also added some sort fields, as well as tests for each of the new components.
**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: gsa/src/web/components/powerfilter/__tests__/cvssbasegroup.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    const handleChange = jest.fn();
+    const {element} = render(
+      <CvssBaseGroup filter={filter} name="name" onChange={handleChange} />,
+    );
+    const input = element.querySelectorAll('input');
+
+    expect(input[0]).toHaveAttribute('value', '7');
+  });
+
+  test('should render value from cvss_base by default', () => {
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/cvssbasegroup.js]:
```suggestion
  test('should render value from cvss prop by default', () => {
```

File: gsa/src/web/components/powerfilter/__tests__/hostgroup.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  });
+
+  test('should render value from host by default', () => {
+    const filter = Filter.fromString('host=foo');
+    const handleChange = jest.fn();
+    const {element} = render(
+      <HostGroup
+        filter={filter}
+        name="name"
+        host={'bar'}
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/hostgroup.js]:
```suggestion
        host="bar"
```

File: gsa/src/web/components/powerfilter/__tests__/hostgroup.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  });
+
+  test('should call change handler', () => {
+    const filter = Filter.fromString('host=foo');
+    const handleChange = jest.fn();
+    const {element} = render(
+      <HostGroup
+        filter={filter}
+        name="name"
+        host={'bar'}
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/hostgroup.js]:
```suggestion
        host="bar"
```

File: gsa/src/web/components/powerfilter/__tests__/locationgroup.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+describe('LocationGroup tests', () => {
+  test('should render', () => {
+    const filter = Filter.fromString('location=general/tcp');
+    const handleChange = jest.fn();
+    const {element} = render(
+      <LocationGroup
+        filter={filter}
+        name="name"
+        loc={'general/tcp'}
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/locationgroup.js]:
```suggestion
        loc="general/tcp"
```

File: gsa/src/web/components/powerfilter/__tests__/locationgroup.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  });
+
+  test('should render value from loc by default', () => {
+    const filter = Filter.fromString('location=general/tcp');
+    const handleChange = jest.fn();
+    const {element} = render(
+      <LocationGroup
+        filter={filter}
+        name="name"
+        loc={'foo'}
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/locationgroup.js]:
```suggestion
        loc="foo"
```

File: gsa/src/web/components/powerfilter/__tests__/locationgroup.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  });
+
+  test('should call change handler', () => {
+    const filter = Filter.fromString('location=general/tcp');
+    const handleChange = jest.fn();
+    const {element} = render(
+      <LocationGroup
+        filter={filter}
+        name="name"
+        loc={'bar'}
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/locationgroup.js]:
```suggestion
        loc="bar"
```

File: gsa/src/web/components/powerfilter/__tests__/ownergroup.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+describe('OwnerGroup tests', () => {
+  test('should render', () => {
+    const filter = Filter.fromString('owner=foo');
+    const handleChange = jest.fn();
+    const {element} = render(
+      <OwnerGroup
+        filter={filter}
+        name="name"
+        owner={'foo'}
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/ownergroup.js]:
```suggestion
        owner="foo"
```

File: gsa/src/web/components/powerfilter/__tests__/ownergroup.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  });
+
+  test('should render value from owner by default', () => {
+    const filter = Filter.fromString('owner=admin');
+    const handleChange = jest.fn();
+    const {element} = render(
+      <OwnerGroup
+        filter={filter}
+        name="name"
+        owner={'foo'}
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/ownergroup.js]:
```suggestion
        owner="foo"
```

File: gsa/src/web/components/powerfilter/__tests__/ownergroup.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+  });
+
+  test('should call change handler', () => {
+    const filter = Filter.fromString('owner=admin');
+    const handleChange = jest.fn();
+    const {element} = render(
+      <OwnerGroup
+        filter={filter}
+        name="name"
+        owner={'bar'}
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/ownergroup.js]:
```suggestion
        owner="bar"
```

File: gsa/src/web/components/powerfilter/__tests__/vulnerabilitygroup.js [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+describe('VulnerabilityGroup tests', () => {
+  test('should render', () => {
+    const filter = Filter.fromString('vulnerability=foo');
+    const handleChange = jest.fn();
+    const {element} = render(
+      <VulnerabilityGroup
+        filter={filter}
+        name="name"
+        vuln={'foo'}
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/vulnerabilitygroup.js]:
```suggestion
        vuln="foo"
```

File: gsa/src/web/components/powerfilter/__tests__/vulnerabilitygroup.js [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+  });
+
+  test('should render value from vulnerability by default', () => {
+    const filter = Filter.fromString('vulnerability=foo');
+    const handleChange = jest.fn();
+    const {element} = render(
+      <VulnerabilityGroup
+        filter={filter}
+        name="name"
+        vuln={'bar'}
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/vulnerabilitygroup.js]:
```suggestion
        vuln="bar"
```

File: gsa/src/web/components/powerfilter/__tests__/vulnerabilitygroup.js [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+  });
+
+  test('should call change handler', () => {
+    const filter = Filter.fromString('vulnerability=foo');
+    const handleChange = jest.fn();
+    const {element} = render(
+      <VulnerabilityGroup
+        filter={filter}
+        name="name"
+        vuln={'bar'}
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/__tests__/vulnerabilitygroup.js]:
```suggestion
        vuln="bar"
```

File: gsa/src/web/components/powerfilter/cvssbasegroup.js [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+import Spinner from '../form/spinner.js';
+import {parseSeverity} from 'gmp/parser.js';
+
+const CvssBaseGroup = ({cvss, filter, onChange, name = 'cvss_base'}) => {
+  if (!isDefined(cvss) && isDefined(filter)) {
+    cvss = parseSeverity(filter.get('cvss_base'));
+  }
+
+  return (
+    <FormGroup title={_('CVSS Base Score')}>
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/cvssbasegroup.js]:
```suggestion
    <FormGroup title={_('Severity')}>
```

File: gsa/src/web/components/powerfilter/filterdialog.js [hunk 14]
```diff
@@ -118,6 +120,13 @@ class FilterDialog extends React.Component {
     this.setState({filter});
   }
 
+  onSearchTermChange(value, name, relation = '~') {
+    const {filter} = this.state;
+    filter.set(name, value.toString(), relation);
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/filterdialog.js]:
```suggestion
    filter.set(name, `${value}`, relation);
```

File: gsa/src/web/components/powerfilter/hostgroup.js [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+
+import FormGroup from '../form/formgroup.js';
+import TextField from '../form/textfield.js';
+
+const HostGroup = ({host, filter, onChange, name = 'host'}) => {
+  if (!isDefined(host) && isDefined(filter)) {
+    host = filter.get('host');
+  }
+  return (
+    <FormGroup title={_('Host IP')}>
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/hostgroup.js]: Not sure if it just filters the IP address or the hostname too. Could you check that?

File: gsa/src/web/pages/results/filterdialog.js [hunk 16]
```diff
@@ -112,10 +126,20 @@ const ResultsFilterDialogComponent = ({
...
+
+    <HostGroup filter={filter} onChange={onSearchTermChange} />
+
+    <LocationGroup filter={filter} onChange={onSearchTermChange} />
+
     <FirstResultGroup filter={filter} onChange={onFilterValueChange} />
 
     <ResultsPerPageGroup filter={filter} onChange={onFilterValueChange} />
 
+    <CvssBaseGroup filter={filter} onChange={onFilterValueChange} />
```
Comment (bjoernricks) [inline gsa/src/web/pages/results/filterdialog.js]: Can we move the cvssbasgroup below the severity class?

File: gsa/src/web/components/powerfilter/filterdialog.js [hunk 17]
```diff
@@ -82,22 +83,23 @@ class FilterDialog extends React.Component {
   }
 
   handleSave() {
-    let {filter, filterName = '', filterstring, saveNamedFilter} = this.state;
+    const {filter, filterName = '', filterstring, saveNamedFilter} = this.state;
     const {onFilterChanged, onCloseClick} = this.props;
 
-    filter = Filter.fromString(filterstring, filter);
+    const newFilter = Filter.fromString(filterstring);
+    newFilter._merge(filter);
```
Comment (bjoernricks) [inline gsa/src/web/components/powerfilter/filterdialog.js]: Would be nice if we can rename the Filter model `_merge` method into `merge` too. Could be done in another PR.

Comment (bjoernricks) [reply gsa/src/web/components/powerfilter/filterdialog.js]: I've used the python scheme for method names where an underscore marks internal methods, which shouldn't be used from the outside of the module. This isn't valid for _merge anymore :-)

File: gsa/src/web/components/powerfilter/hostgroup.js (hunk 15 above)
Comment (saberlynx) [reply gsa/src/web/components/powerfilter/hostgroup.js]: I think it's just the host IP. In the API Documentation there is a separate "hostname" keyword for the latter.

File: gsa/src/web/pages/results/filterdialog.js (hunk 16 above)
Comment (saberlynx) [reply gsa/src/web/pages/results/filterdialog.js]: Will do :)

File: gsa/src/web/components/powerfilter/filterdialog.js (hunk 17 above)
Comment (saberlynx) [reply gsa/src/web/components/powerfilter/filterdialog.js]: Cool. I'll make another PR for that.

**Pull request merged**